alacritty_terminal/src/input.rs

```diff
@@ -930,7 +930,10 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
                 // Don't suppress when there has been a `ReceiveChar` action
                 Some(suppress_chars.unwrap_or(true) && binding.action != Action::ReceiveChar)
             })
-            .unwrap_or(false);
+            .unwrap_or_else(|| {
+                // Do not suppress char if no bindings were triggered
+                false
+            });
```

Comment:
We shouldn't use `unwrap_or_else` here. You can just put the comment above the `unwrap_or`.

The `unwrap_or_else` method should only be used with non-constant values that are calculated at runtime.

Reply:
Won't it be optimized out?

Reply:
Relying on optimizer, when things can be easily changed to a faster code is a bad idea.

Reply:
Seemed so trivial. I just used it to match style with above lines, but should be corrected new as you requested.

Reply:
Just as a sidenote, you're both right here.

For constant values, both will result in exactly the same code in every case as far as I remember. The difference is just that cases which are not constant benefit from `_or_else` in the case where the value is `Some`.

However, if we have the option to express something properly, we should do so. This is simply a style issue and using a function meant for non-constant values to create a closure which just returns a constant simply isn't a good code style.